### ISSUE TITLE
[bug] Macos screenshare permission

### ISSUE THREAD

**Issue by nicolas-stephan** [tauri-apps/tauri#7647]
### Describe the bug

My application uses a rust library for screensharing (https://docs.rs/scrap/latest/scrap/). When the user needs to screen share, they first have to enable the permission in MacOS, and once enabled, everything is fine. Until it gets updated. As soon as the application is updated, the user can no longer screen share, although the permissions in MacOS remain the same. To get the permission to work again, you have to do some strange manipulation.

### Reproduction

Download the  project (https://github.com/holoffice/screenshare-macos-permission)

- Build the application (yarn tauri build).
- Move the application to your Mac's Applications folder and launch it.
- Try taking a screenshot to give the application permission to capture.
- Restart to make sure you now can take a screenshot.
- Remove the application from the applications folder.
- Update "tauri.conf.json" version from 0.0.0 to 0.0.1.
- Rebuild the application.
- Move the application to your Mac's Applications folder and launch it.
- Try taking a screenshot. This time it will ask you for permission again, even though you have already given it.

The only way to take a screenshot is to remove the application from the permissions section and start it again.

### Expected behavior

The permission should still have screenshare permission after an update

### Platform and versions

```text
[✔] Environment
    - OS: Mac OS 13.0.0 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.69.0 (84c898d65 2023-04-16)
    ✔ Cargo: 1.69.0 (6e9a83356 2023-04-12)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-apple-darwin (default)
    - node: 16.19.1
    - yarn: 1.22.19
    - npm: 8.19.3

[-] Packages
    - tauri [RUST]: 1.4.1
    - tauri-build [RUST]: 1.4.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: 1.4.0
    - @tauri-apps/cli [NPM]: 1.4.0

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

**Comment by wyhaya** [comment]
@nicolas-stephan Is your app signed? If not, I would think it would be due to unsigned, every time you release a new version, macOS will consider it a new application and will ask the user to re-license it.

I recently had a similar problem with my app using a keychain, every time I ran the app during development I was asked to license the keychain permissions, after I signed it and uploaded it to Apple for notarization everything worked fine.
